spimdisasm/mips/symbols/MipsSymbolBase.py

```diff
@@ -193,36 +193,32 @@ def shouldMigrate(self) -> bool:
         return False
 
 
-    def renameBasedOnType(self):
-        if not common.GlobalConfig.AUTOGENERATED_NAMES_BASED_ON_DATA_TYPE:
-            return
-
+    def autogenerateName(self, contextSym, inSymbolOffset):
         if not self.contextSym.isAutogenerated:
             return
 
-        if self.isFloat(0):
-            self.contextSym.name = f"FLT_{self.vram:08X}"
-        elif self.isDouble(0):
-            self.contextSym.name = f"DBL_{self.vram:08X}"
-        elif self.isString():
-            self.contextSym.name = f"STR_{self.vram:08X}"
-        elif self.isPascalString():
-            self.contextSym.name = f"PSTR_{self.vram:08X}"
-
+        if common.GlobalConfig.AUTOGENERATED_NAMES_BASED_ON_FILE_NAME and self.parent is not None:
+            offset = self.inFileOffset + inSymbolOffset
+            contextSym.name = f"{self.parent.name}{self.sectionType.toStr()}_{offset:06X}"
+        elif common.GlobalConfig.AUTOGENERATED_NAMES_BASED_ON_DATA_TYPE:
+            if self.isFloat(0):
+                contextSym.name = f"FLT_{self.vram:08X}"
+            elif self.isDouble(0):
+                contextSym.name = f"DBL_{self.vram:08X}"
+            elif self.isString():
+                contextSym.name = f"STR_{self.vram:08X}"
+            elif self.isPascalString():
+                contextSym.name = f"PSTR_{self.vram:08X}"
```

Comment:
To be honest this should be handled in `ContextSymbol.getDefaultName` instead of here, but that should have been fixed a long time ago and I don't expect you to fix this technical debt

Reply:
Alright I'll leave it for now, sorry

Reply:
It's fine, it isn't your fault lol